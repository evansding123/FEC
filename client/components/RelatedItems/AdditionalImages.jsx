import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

const AdditionalImages = ({images, changeMainImage}) => {

  const [state, setState] = useState({idx: 0, displayed: [...images].slice(0, 4)});

  const toggleMainImage = (e) => {
    changeMainImage(e.target.src);
  }

  return (
    <div className="additional-images">
      <button className="slideLeft-mini"
        onClick={()=>setState({idx: state.idx-1, displayed: [...images].slice(state.idx-1, state.idx+3)})}
        hidden={state.idx === 0}
        data-testid="slideLeft-mini">
          &lt;
        </button>
      {state.displayed.map((image, i)=><img data-testid={i} key={image} className="thumbnail" src={image} onClick={toggleMainImage}/>)}
      <button className="slideRight-mini"
          onClick={()=>setState({idx: state.idx+1, displayed: [...images].slice(state.idx+1, state.idx+5)})}
          disabled={images.length < 4 || state.idx === images.length-4}
          data-testid="slideRight-mini">
            &gt;
      </button>
    </div>
  )
}

AdditionalImages.propTypes = {
  images: PropTypes.array,
  changeMainImage: PropTypes.func
}

export default AdditionalImages;