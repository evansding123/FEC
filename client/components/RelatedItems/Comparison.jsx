import React from 'react';
import axios from 'axios';
import DescirptionRow from './DescriptionRow.jsx';
import PropTypes from 'prop-types';

class Comparison extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      featureList: []
    };
    this.handleClick = this.handleClick.bind(this);
    this.combineAllFeatures = this.combineAllFeatures.bind(this);
  }

  // performance wise - I need to fetch data for currently displayed product at a different place
  componentDidMount() {
    axios.get('/get', {params: {endpoint: `products/${this.props.id}`}})
      .then((results) => {
        this.setState({
          currentProduct: results.data
        }, this.combineAllFeatures)
      })
      .catch((err) => {
        console.log(err);
      })
  }

  combineAllFeatures() {
    var allFeatureObjs = this.props.product.features.concat(this.state.currentProduct.features);
    var allFeatures = allFeatureObjs.map(obj=>obj.feature);
    // filter out the duplicate features
    var featureList = [...new Set(allFeatures)];
    this.setState({
      featureList: featureList
    })
  }

  handleClick() {
    this.props.togglePop();
  }

  render() {
    const{product} = this.props;
    const {currentProduct, featureList} = this.state;
    return (
      <div className="modal">
        <div className="modal_content">
          <span className="close" onClick={this.handleClick}>&times;</span>
          <p>Comparing</p>
          <table>
            <thead>
              <tr className="column">
                <th className="col-1">{product.name}</th>
                <th className="col-2"></th>
                <th className="col-3">{!currentProduct ? null: currentProduct.name}</th>
              </tr>
            </thead>
            <tbody className="description-row">
              {featureList.map(feature=>
                <DescirptionRow key={feature} feature={feature} relatedProduct={product.features} currentProduct={currentProduct.features}/>
              )}
            </tbody>
          </table>
        </div>
      </div>
    )
  }
}

Comparison.propTypes = {
  id: PropTypes.number,
  togglePop: PropTypes.func,
  product: PropTypes.object
}

export default Comparison;
