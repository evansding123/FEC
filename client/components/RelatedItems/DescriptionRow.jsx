import React from 'react';
import PropTypes from 'prop-types';

const DescriptionRow = ({feature, relatedProduct, currentItem}) => {
  // get an array of obj (if it does not contain a particular feature array should be empty)
  var featureInRelatedProduct = relatedProduct.filter(entry=>entry.feature===feature);
  var featureInCurrentProduct = currentItem.filter(entry=>entry.feature===feature);
  var displayCol1 = null;
  var displayCol3 = null;
  if (featureInRelatedProduct.length) {
    // the current product has a feature, and therefore check its value
    if (featureInRelatedProduct[0].value) {
      displayCol1 = featureInRelatedProduct[0].value;
    } else {
      // used JS source code
      displayCol1 = "\u2713";
    }
  }
  if (featureInCurrentProduct.length) {
    // the current product has a feature, and therefore check its value
    if (featureInCurrentProduct[0].value) {
      displayCol3 = featureInCurrentProduct[0].value;
    } else {
      displayCol3 = "\u2713";
    }
  }
  return (
    <tr className="row">
      <td className="col-1">{displayCol1}</td>
      <td className="col-1">{feature}</td>
      <td className="col-3">{displayCol3}</td>
    </tr>
  )
}

DescriptionRow.propTypes = {
  feature: PropTypes.string,
  relatedProduct: PropTypes.array,
  currentItem: PropTypes.array
}

export default DescriptionRow;