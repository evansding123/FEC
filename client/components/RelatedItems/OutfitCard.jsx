import React from 'react';
import Price from './Price.jsx';
import Rating from './Rating.jsx';
import PropTypes from 'prop-types';

const OutfitCard = ({product, removeFromOutfit}) => {
  return (
    <li className="card outfit">
      <div className="parent">
        <span className="close" onClick={()=>removeFromOutfit(product.id)}>&#9447;</span>
        <img className="related-product-img" src={product.results[0].photos[0].url}/>
        <div className="product-info">
          <div>{product.category}</div>
          <div className="product-name">{product.name}</div>
          <Price product={product}/>
          <Rating rating={product.rating.ratings} />
        </div>
      </div>
    </li>
  )
}

OutfitCard.propTypes = {
  removeFromOutfit: PropTypes.func,
  product: PropTypes.object
}

export default OutfitCard;