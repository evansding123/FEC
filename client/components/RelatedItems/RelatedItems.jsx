import React from 'react';
import RelatedProducts from './RelatedProducts.jsx';
import Outfits from './Outfits.jsx';
import axios from 'axios';
import PropTypes from 'prop-types';

class RelatedItems extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      relatedItems: [], // related products IDs
      relatedItemsList: [],
      selectedItemsList: [],
      selected: false // if the current product is added to outfit list
    }
    this.getRelatedItemsIds = this.getRelatedItemsIds.bind(this);
    this.renderRelatedItems = this.renderRelatedItems.bind(this);
    this.getAllProductInfo = this.getAllProductInfo.bind(this);
    this.addToOutfit = this.addToOutfit.bind(this);
    this.removeFromOutfit = this.removeFromOutfit.bind(this);
  }

  componentDidMount() {
    this.getRelatedItemsIds(this.props.id);
  }

  componentDidUpdate(prevProps) {
    if (prevProps.id !== this.props.id) {
      this.getRelatedItemsIds(this.props.id);
    }
  }

  getRelatedItemsIds (id) {
    axios.get('/get', {params: {endpoint: `products/${id}/related`}})
      .then((response) =>{
        this.setState({
          relatedItems: response.data,
          selected: false // everytime different product is displayed, change selected back to false
        }, this.renderRelatedItems)
      })
      .catch(err=>{
        console.log(err)
      });
  }

  renderRelatedItems() {
    var promises = [];
    for (var i = 0; i < this.state.relatedItems.length; i++) {
      promises.push(this.getAllProductInfo(this.state.relatedItems[i]));
    }
    Promise.all(promises)
      .then((response) => {
        this.setState({
          relatedItemsList: response
        })
      })
      .catch((err) => {
        console.log(err);
      })
  }

  getAllProductInfo(id) {
    return new Promise((resolve, reject) => {
      axios.get('/get', {params: {endpoint: `products/${id}`}})
        .then((product) => {
          axios.get('/get', {params: {endpoint: `products/${id}/styles`}})
            .then((styles) => {
              axios.get('/get', {params: {endpoint: `reviews/meta/?product_id=${id}`}})
                .then((rating) => {
                  var mergedList = Object.assign(product.data, styles.data)
                  mergedList['rating'] = rating.data
                  resolve(mergedList)
                })
            })
        })
        .catch((err) => {
          reject(err);
        })
    })
  }

  addToOutfit() {
    if (!this.state.selected) {
      this.getAllProductInfo(this.props.id)
        .then((results) => {
          this.setState({
            selectedItemsList: [results, ...this.state.selectedItemsList],
            selected: true
          })
        })
        .catch((err) => {
          console.log(err);
        })
    }
  }

  removeFromOutfit(id) {
    var updated = [...this.state.selectedItemsList];
    for (var i = 0; i < updated.length; i++) {
      if (updated[i].id === id) {
        updated.splice(i, 1);
      }
    }
    this.setState({
      selectedItemsList: updated,
      selected: false
    })
  }

  render() {
    const {id, changeProductId} = this.props;
    const {relatedItemsList, selectedItemsList} = this.state;
    return (
      <div>
        { relatedItemsList.length !== 0 ?
        <>
          <RelatedProducts id={id} relatedItemsList={relatedItemsList} changeProductId={changeProductId}/>
          <Outfits selectedItemsList={selectedItemsList} addToOutfit={this.addToOutfit} removeFromOutfit={this.removeFromOutfit}/>
        </> : null }
      </div>
    )
  }
}

RelatedItems.propTypes = {
  id: PropTypes.number,
  changeProductId: PropTypes.func
}

export default RelatedItems;