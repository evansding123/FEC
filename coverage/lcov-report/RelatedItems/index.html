
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for RelatedItems</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> RelatedItems</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
<<<<<<< HEAD
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>4/4</span>
=======
                <span class="strong">62.06% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>247/398</span>
>>>>>>> main
            </div>
        
            
            <div class='fl pad1y space-right2'>
<<<<<<< HEAD
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>1/1</span>
=======
                <span class="strong">84.21% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>16/19</span>
>>>>>>> main
            </div>
        
            
            <div class='fl pad1y space-right2'>
<<<<<<< HEAD
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>1/1</span>
=======
                <span class="strong">39.29% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>11/28</span>
>>>>>>> main
            </div>
        
            
            <div class='fl pad1y space-right2'>
<<<<<<< HEAD
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>4/4</span>
=======
                <span class="strong">62.06% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>247/398</span>
>>>>>>> main
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
<<<<<<< HEAD
    <div class='status-line high'></div>
=======
    <div class='status-line medium'></div>
>>>>>>> main
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
<<<<<<< HEAD
	<td class="file high" data-value="sampleTest.jsx"><a href="sampleTest.jsx.html">sampleTest.jsx</a></td>
=======
	<td class="file low" data-value="AdditionalImages.jsx"><a href="AdditionalImages.jsx.html">AdditionalImages.jsx</a></td>
	<td data-value="34.85" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 34%"></div><div class="cover-empty" style="width: 66%"></div></div>
	</td>
	<td data-value="34.85" class="pct low">34.85%</td>
	<td data-value="66" class="abs low">23/66</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="34.85" class="pct low">34.85%</td>
	<td data-value="66" class="abs low">23/66</td>
	</tr>

<tr>
	<td class="file low" data-value="Comparison.jsx"><a href="Comparison.jsx.html">Comparison.jsx</a></td>
	<td data-value="32.47" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 32%"></div><div class="cover-empty" style="width: 68%"></div></div>
	</td>
	<td data-value="32.47" class="pct low">32.47%</td>
	<td data-value="77" class="abs low">25/77</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="32.47" class="pct low">32.47%</td>
	<td data-value="77" class="abs low">25/77</td>
	</tr>

<tr>
	<td class="file low" data-value="DescriptionRow.jsx"><a href="DescriptionRow.jsx.html">DescriptionRow.jsx</a></td>
	<td data-value="28.57" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 28%"></div><div class="cover-empty" style="width: 72%"></div></div>
	</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="42" class="abs low">12/42</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="42" class="abs low">12/42</td>
	</tr>

<tr>
	<td class="file high" data-value="Price.jsx"><a href="Price.jsx.html">Price.jsx</a></td>
>>>>>>> main
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
<<<<<<< HEAD
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
=======
	<td data-value="18" class="abs high">18/18</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="18" class="abs high">18/18</td>
	</tr>

<tr>
	<td class="file high" data-value="Rating.jsx"><a href="Rating.jsx.html">Rating.jsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

<tr>
	<td class="file medium" data-value="RelatedProductCard.jsx"><a href="RelatedProductCard.jsx.html">RelatedProductCard.jsx</a></td>
	<td data-value="76.6" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 76%"></div><div class="cover-empty" style="width: 24%"></div></div>
	</td>
	<td data-value="76.6" class="pct medium">76.6%</td>
	<td data-value="94" class="abs medium">72/94</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="37.5" class="pct low">37.5%</td>
	<td data-value="8" class="abs low">3/8</td>
	<td data-value="76.6" class="pct medium">76.6%</td>
	<td data-value="94" class="abs medium">72/94</td>
	</tr>

<tr>
	<td class="file high" data-value="RelatedProducts.jsx"><a href="RelatedProducts.jsx.html">RelatedProducts.jsx</a></td>
	<td data-value="94.94" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 94%"></div><div class="cover-empty" style="width: 6%"></div></div>
	</td>
	<td data-value="94.94" class="pct high">94.94%</td>
	<td data-value="79" class="abs high">75/79</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="94.94" class="pct high">94.94%</td>
	<td data-value="79" class="abs high">75/79</td>
>>>>>>> main
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
<<<<<<< HEAD
                at Thu Jun 10 2021 17:25:21 GMT-0700 (Pacific Daylight Time)
=======
<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
                at Wed Jun 09 2021 00:12:45 GMT-0700 (Pacific Daylight Time)
=======
                at Thu Jun 10 2021 10:38:30 GMT-0700 (Pacific Daylight Time)
>>>>>>> main
=======
                at Thu Jun 10 2021 10:52:40 GMT-0700 (Pacific Daylight Time)
>>>>>>> main
=======
                at Thu Jun 10 2021 10:52:40 GMT-0700 (Pacific Daylight Time)
=======
                at Thu Jun 10 2021 10:52:34 GMT-0700 (Pacific Daylight Time)
>>>>>>> f003810a1a7c451f7e70b6b1598f6ac11b850853
>>>>>>> main
>>>>>>> origin
            </div>
        </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    