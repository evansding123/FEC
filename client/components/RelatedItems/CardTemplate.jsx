import React from 'react';
import Rating from './Rating.jsx';
import Comparison from './Comparison.jsx';
import Price from './Price.jsx';
import AdditionalImages from './AdditionalImages.jsx';
import PropTypes from 'prop-types';

class CardTemplate extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      seen: false, // whether a modal is displayed or not
      display: false, // whether additional images are displayed or not
      additionalImages: []
    }
    this.togglePop = this.togglePop.bind(this);
    this.displayAdditionalImages = this.displayAdditionalImages.bind(this);
    this.hideAdditionalImages = this.hideAdditionalImages.bind(this);
  }

  togglePop() {
    this.setState({
      seen: !this.state.seen
    })
  }

  displayAdditionalImages() {
    var photosArr = this.props.product.results.map(style=>style.photos[0].thumbnail_url);
    this.setState({
      display: true,
      additionalImages: photosArr
    })
  }

  hideAdditionalImages() {
    this.setState({
      display: false
    })
  }

  render() {
    const {id, product, deselect} = this.props;
    if (id === "outfit") {
      // if it is an outfit card
      return (
        <li className="card outfit">
          <div className="parent">
            <span className="close" onClick={()=>deselect()}>&#9447;</span>
            <img className="related-product-img" src={product.results[0].photos[0].url}/>
            <div className="product-info">
              <div>{product.category}</div>
              <div className="product-name">{product.name}</div>
              <Price product={product}/>
              <Rating rating={product.rating.ratings} />
            </div>
          </div>
        </li>
      )
    } else {
      // else it is an related item card
      return (
        <li className="card">
          <div className="parent">
            {this.state.seen ?
            <Comparison togglePop={this.togglePop} product={product} id={id}/> : null}
            <span className="star" onClick={this.togglePop}>&#9734;</span>
            <img className="related-product-img" src={product.results[0].photos[0].url} onMouseOver={this.displayAdditionalImages} onMouseOut={this.hideAdditionalImages}/>
            {!this.state.display ? null: <AdditionalImages images={this.state.additionalImages}/>}
            <div className="product-info">
              <div className="product-category">{product.category}</div>
              <div className="product-name">{product.name}</div>
              <Price product={product}/>
              <Rating rating={product.rating.ratings} />
            </div>
          </div>
        </li>
      )
    }
  }
}

CardTemplate.propTypes = {
  id: PropTypes.number,
  deselect: PropTypes.func,
  product: PropTypes.object
}

export default CardTemplate;