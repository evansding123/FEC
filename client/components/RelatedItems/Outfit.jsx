import React from 'react';
import CardTemplate from './CardTemplate.jsx';

const Outfit = (props) => {
  return (
    <div className="container">
      <h3 className="related-products">Outfit</h3>
      <div className="carousel" data-orientation="horizontal" data-scrollable="true">
        <ul className="slider">
          {props.selectedItemsList.map(product=>
            <CardTemplate product={product} />
            )}
        </ul>
      </div>
    </div>
  )
}

export default Outfit;