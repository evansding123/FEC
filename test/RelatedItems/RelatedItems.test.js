/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, waitForElement, fireEvent, cleanup, screen } from '@testing-library/react';
import RelatedItems from '../../client/components/RelatedItems/RelatedItems.jsx';
import axiosMock from 'axios';
import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup)

it('should render the RelatedItems component', () => {
  const { asFragment } = render(<RelatedItems id={25811}/>);
  expect(asFragment(<RelatedItems id={25811}/>)).toMatchSnapshot()
})

it('should render the RelatedItems component', () => {
  render(<RelatedItems id={25811}/>);
  const relatedItems = [1, 2, 3, 4, 5];
  expect(screen.getByTestId("card")).toHaveTextContent("1")
})