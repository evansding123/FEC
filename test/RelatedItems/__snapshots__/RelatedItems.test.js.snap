// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the RelatedItems component 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
